pkg/ocmprovider/ocmprovider.go

```diff
@@ -150,21 +149,33 @@ func (s *ocmProvider) Get() ([]upgradev1alpha1.UpgradeConfigSpec, error) {
 		return nil, ErrMissingChannelGroup
 	}
 
-	// Retrieve the cluster's upgrade policy from Cluster Services
+	// Retrieve the cluster's available upgrade policies from Cluster Services
 	upgradePolicies, err := getClusterUpgradePolicies(cluster, s.httpClient, s.ocmBaseUrl)
 	if err != nil {
 		log.Error(err, "error retrieving upgrade policies")
 		return nil, ErrRetrievingPolicies
 	}
 
-	// Apply the Upgrade Policies to the cluster
-	specs, err := buildUpgradeConfigSpecs(upgradePolicies, cluster, s.client)
-	if err != nil {
-		log.Error(err, "cannot build UpgradeConfigs from policy")
-		return nil, ErrProcessingPolicies
-	}
+	// Get the next occuring policy from the available policies
+	if len(upgradePolicies.Items) > 0 {
```

Comment:
I haven't tested this to be sure, but if there are no policies returned, this will return nil and nil, which will break the upgradeconfigmanager. If nil is returned as the error, it expects a proper list of UpgradeConfigSpecs ([link](https://github.com/openshift/managed-upgrade-operator/blob/1f0d78c7b314c1d2008461823535b672349408c7/pkg/upgradeconfigmanager/upgradeconfigmanager.go#L184)). Can you please test this scenario and make sure it works ok?

Reply:
(And if this is indeed an issue, it means we need a test for it, because evidently our existing ones didn't catch it :/)

Reply:
This is actually a result of BDD already, I had to add the above to account for 0 upgradePolicies. Also it appears that an empty slice should be returned only when encoding to JSON, otherwise a `nil` slice is fine: https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices